packages/react-card/Spec.md

```diff
@@ -0,0 +1,403 @@
+
+Keyboard interaction
+
+1. Focus Initial focus is set on card, to access items within the card use EnterKey or Spacebar
+2. Focus-Inside Focus moves to first actionable item in the card. Use ArrowKeys or TabKey to move focus to child elements within the card. To exit card use EscKey, to move focus back to card [parent]
+3. Checkbox-Focus Depending on your tab order of items within the card, the number of keystrokes will vary. Once focus is set on the checkbox by default it will be set to unselected
+4. SpaceBar or EnterKey will toggle both Card and Checkbox to selected state. Keying again will toggle to unselected
+
+### Context menu
+
+![Visual interaction of the context menu on cards using the keyboard](./assets/context-interaction-keyboard.png)
+
+1. Rest
+2. OnFocus - Focus rect. is set on card. Use shiftKey and press F10 to invoke context menu
+
+   `aria-haspopup="menu"`
+
+   `aria-expanded="false"`
+
+3. ShowSet focus on context menu
+
+   `aria-expanded="true"`
+
+4. Hide/Dismiss - Use EscKey to hide/dismiss
+
+   `aria-expanded="false"`
+
+## Accessibility
+
+> ⚠️ Note: All other accessibility information, not covered in this section, is provided throughout the spec.
+
+Each card is of `role="group"`.
+
+### Narrator guidance
+
+#### Selectable cards
+
+![Visual interaction of keyboard usage with selectable cards](./assets/selectable-narrator.png)
+
+1 Focus + unselected
+
+1. Label: Select this card
+2. Role: Group
+3. State: Focus + unselected
+4. Helper text: Use enterKey or spacebar to toggle state to selected
+
+2 Focus + selected
+
+1. Label: Unselect this card
+2. Role: Group
+3. State: Focus + selected
+4. Helper text: Use enterKey or spacebar to toggle state to unselected
+
+#### Non-interactive cards
+
+![Visual interaction of keyboard usage with non-interactive cards](./assets/non-interactive-narrator.png)
+
+1 Focus + unselected
+
+1. Label: App Card
+2. Role: Group
+3. State: Focus
+4. Helper text: Use enterKey to access controls within the card
+
+2 Focus (inside)
+
+1. Label: Open App
+2. Role: Button
+3. State: Focus
+4. Helper text: Use enterkey to open app in new window
+
+3 Focus Checkbox unselected
+
+1. Label: Share App
+2. Role: Button
+3. State: Focus
+4. Helper text: Use enter key to open share dialog
+
+4 Focus Checkbox selected
+
+1. Label: Select card
+2. Role: Checkbox
+3. State: Unselected
+4. Helper text: Use spacebar or enter key to select this item
+
+#### Disabled
+
+![Two disabled cards. One is unselected and one is selected](./assets/disabled-narrator.png)
+
+1 Unselected disabled
+
+1. Label: Selectable Card
+2. Role: Group
+3. State: Disabled, unselected
+
+2 Selected disabled
+
+1. Label: Selectable Card
+2. Role: Group
+3. State: Disabled, selected
+
+#### Context menu
+
+![Context menu opening and closing](./assets/context-narrator.png)
+
+RightClick / ShiftKey + F10
+
+1. OnFocus
+
+   `aria-haspopup="true"`
+
+   `aria-expanded="false"`
+
+   `aria-controls="menu"`
```

Comment:
usually menu appears in DOM only after it is triggered, so it is not possible to reference it in aria-controls before opening.

Reply:
Is there a suggestion or should this just be removed?